CI setup that runs tests e2e testing using GPU machines

mathfac opened the pull request:
## Motivation and Context
CI setup that runs tests using GPU machines:
- Builds and installs habitat-sim with ` python setup.py install` and `./build.sh --headless`
- Runs `pytest` tests on habitat-sim with testing data
- Added speed regression test with FPS threshold (900 FPS) , thanks @erikwijmans for advice how to do it in elegant way.
- Added run of habitat-api tests on master branch. That will guarantee that HSIM changes won't break HAPI
- Added pytorch installation and running habitat baseline test in prediction mode. That maybe too much of testing, but decided to keep it for now.

Have ideas how to speed  up and parallelize jobs. Will follow up in next PRs.
<!--- Why is this change required? What problem does it solve? -->
<!--- Please link to an existing issue here if one exists. -->
<!--- (we recommend to have an existing issue for each pull request) -->

## How Has This Been Tested
Tests time estimations:
```
Install conda and dependencies 02:56
Install cuda 04:32
Build, install habitat-sim and run benchmark 03:13
Run sim tests 03:08
Run api tests with pytorch 08:14
```

<!--- Please describe here how your modifications have been tested. -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

erikwijmans commented on .circleci/config.yml:
```diff
@@ -0,0 +1,89 @@
+              export PATH=$HOME/miniconda/bin:$PATH
+              conda create -y -n habitat python=3.6
+              . activate habitat
+              conda install -y numpy pyyaml scipy ipython mkl mkl-include pytest
```
Let's add a `conda install -y -c conda-forge ninja` -- This should help with build speed

erikwijmans commented on .circleci/config.yml:
```diff
@@ -0,0 +1,89 @@
+              python examples/example.py --scene data/scene_datasets/habitat-test-scenes/van-gogh-room.glb
+              export PYTHONPATH=$PYTHONPATH:$(pwd)
+              ./build.sh --headless
+              pip install -e .
```
Possibly bad things will happen as the pip install isn't headless.

```suggestion
              HEADLESS=True pip install -e .
```

msavva commented on .circleci/config.yml:
```diff
@@ -0,0 +1,89 @@
+              sudo apt-get --yes --force-yes install cuda
+              nvidia-smi
+      - run:
+          name: Install habtiat-sim
```
typo: `habtiat` -> `habitat`

mathfac commented:
- Added speed regression test, thanks @erikwijmans for advice how to do it in elegant way
- Added run of habitat-api tests on master branch. That will guarantee that HSIM changes won't break HAPI
- Added pytorch installation and running habitat baseline test in prediction mode. That maybe too much of testing, but decided to keep it for now.

mathfac commented:
CI tests passed, merging.
